__tests__/pet.test.js

```diff
@@ -0,0 +1,142 @@
+describe('Hunger', () => {
+    it('has a initial hunger of 10', () => {
+        const pet = new Pet('Fido');
+        expect(pet.hunger).toEqual(0);
+    });
+});
+describe('Fitness', () => {
+    it('has a initial fitness of 10', () => {
+        const pet = new Pet('Fido');
+        expect(pet.fitness).toEqual(10);
+    });
+});
+describe('GrowUp', () => {
+    it('Adds 1 to initial age, adds 5 to initial hunger and reduces fitness by 3', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        expect(pet.age).toEqual(1);
+        expect(pet.hunger).toEqual(5);
+        expect(pet.fitness).toEqual(7);
+    });
+    it('throws an error if the pet is not alive', () => {
+        const petOld = new Pet('Fido');
+        petOld.age = 30;
+        expect(() => petOld.growUp()).toThrow('Your pet is no longer alive :(');
+    });
+});
+describe('walk', () => {
+    it('Adds 4 to fitness to a max of 10', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        pet.feed();
+        pet.growUp();
+        expect(pet.fitness).toEqual(4);
+        pet.walk();
+        expect(pet.fitness).toEqual(8);
+        pet.walk();
+        expect(pet.fitness).toEqual(10);
+    });
+    it('throws an error if the pet is not alive', () => {
+        const petOld = new Pet('Fido');
+        petOld.age = 30;
+        expect(() => petOld.walk()).toThrow('Your pet is no longer alive :(');
+    });
+});
+describe('feed', () => {
+    it('Reduces hunger to a min of 0', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        pet.feed();
+        pet.growUp();
+        expect(pet.hunger).toEqual(7);
+        pet.feed();
+        expect(pet.hunger).toEqual(4);
+        pet.feed();
+        expect(pet.hunger).toEqual(1);
+        pet.feed();
+        expect(pet.hunger).toEqual(0);
+    });
+    it('throws an error if the pet is not alive', () => {
+        const petOld = new Pet('Fido');
+        petOld.age = 30;
+        expect(() => petOld.feed()).toThrow('Your pet is no longer alive :(');
+    });
+});
+describe('checkup', () => {
+    it('Return results of checkup based on Hunger and Fitness', () => {
+        const pet = new Pet('Fido');
+        pet.checkup();
+        expect(pet.checkupResult).toEqual("I feel great")
+        pet.growUp();
+        pet.feed();
+        pet.growUp();
+        pet.feed()
+        pet.growUp();
+        pet.checkup();
+        expect(pet.checkupResult).toEqual("I am hungry AND I need a walk")
+        pet.feed();
+        pet.feed();
+        pet.feed();
+        pet.feed();
+        pet.checkup();
+        expect(pet.checkupResult).toEqual("I need a walk")
+        pet.walk();
+        pet.growUp();
+        pet.walk();
+        pet.checkup();
+        expect(pet.checkupResult).toEqual("I am hungry")
+        pet.feed();
+        pet.feed();
+        pet.feed();
+        pet.walk();
+        pet.checkup();
+        expect(pet.checkupResult).toEqual("I feel great")
+    });
```

Comment:
As above this could be refactored:
```
it("Return results of checkup based on Hunger and Fitness", () => {
        const pet = new Pet("Fido");
        pet.checkup();
        expect(pet.checkupResult).toEqual("I feel great");

        pet.fitness = 2;
        pet.hunger = 8;
        pet.checkup()
        expect(pet.checkupResult).toEqual("I am hungry AND I need a walk");

        pet.fitness = 2;
        pet.hunger = 2;
        pet.checkup();
        expect(pet.checkupResult).toEqual("I need a walk");

        pet.fitness = 8;
        pet.hunger = 8;
        pet.checkup();
        expect(pet.checkupResult).toEqual("I am hungry");
    });
```